encodings/fastlanes/src/bitpacking/mod.rs

```diff
@@ -42,7 +42,12 @@ impl BitPackedArray {
     /// Create a new bitpacked array using a buffer of packed data.
     ///
     /// The packed data should be interpreted as a sequence of values with size `bit_width`.
-    pub fn try_new(
+    ///
+    /// # Safety
+    ///
+    /// If `ptype` is signed, `packed` may **not** contain any values that once unpacked would
+    /// be interpreted as negative.
+    pub unsafe fn try_new(
```

Comment:
API-wise, this feels iffy. There are probably other array types that can get some inputs that create weird states. WDYT about having a `try_new_unchecked` function + a checked function that does some similar validation to what we have here.

Reply:
The thing that is annoying is that this function really shouldn't be `pub` but it has to be so that it's accessible from the sampling compressor.

Otherwise users of the public API should be going through `BitPackedArray::encode`.

My goal with the unsafe was to make it clear that there are invariants that users need to check themselves for the args they're passing to the function. I don't think we can check those invariants here without fully decompressing the packed buffer

Reply:
That makes sense. WDYT about just changing it to `unchecked` to keep the "convention" of `try_new` as something that errors safely everywhere else?

Reply:
yea i can do that